plugins/modules/configuration_item_batch.py

```diff
@@ -0,0 +1,171 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2021, XLAB Steampunk <steampunk@xlab.si>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+module: configuration_item_batch
+
+author:
+  - Manca Bizjak (@mancabizjak)
+  - Miha Dolinar (@mdolin)
+  - Tadej Borovsak (@tadeboro)
+
+short_description: Manage ServiceNow configuration items
+
+description:
+  - Create, delete or update a ServiceNow configuration item.
+  - For more information, refer to the ServiceNow configuration management documentation at
+    U(https://docs.servicenow.com/bundle/paris-servicenow-platform/page/product/configuration-management/reference/cmdb-table-property-descriptions.html).
+version_added: 1.0.0
+extends_documentation_fragment:
+  - servicenow.itsm.instance
+
+seealso:
+  - module: servicenow.itsm.configuration_item_info
+
+options:
+  sys_class_name:
+    description:
+      - The sys_class_name parameter value contains a table name (configuration item type)
+        that we would like to manipulate. In our example, we are manipulating items in the
+        cmdb_ci_ec2_instance table.
+    required: true
+    type: str
+  id_column:
+    description:
+      - The id_column parameter contains a configuration item column name that should be
+        used to determine if we need to update an existing configuration item or if we must
+        create a new one. In the example above, we are using instance ID as a key, but we could
+        also use things like VM names (if they are unique), for example.
+    required: true
+    type: str
```

Comment:
We should generalize this a bit since there may be situations where a single column cannot uniquely identify the CMDB item, but a combination of two or more could. For example, VM names might not be unique, but if we combine them with VPC id, they might become.

```suggestion
  id_column_set:
    description:
      - Columns that should be used to identify an existing record that we need to update.
    required: true
    type: list
    elements: str
```

We should also add a validation that makes sure we have at least one column (that the list is not empty).